File: src/main/java/hyeon9mak/multidatasourcequerycounter/QueryCountLogger.java
package hyeon9mak.multidatasourcequerycounter;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Component;

@Component
public class QueryCountLogger {

    private static final String LOG_MESSAGE_FORMAT = "'{}' - totalQueryCount: {}, totalSpendTime: {}ms";

    private final Logger logger = LoggerFactory.getLogger(getClass());
    private final QueryCounterLoggingLevelProperties properties;

    public QueryCountLogger(QueryCounterLoggingLevelProperties properties) {
        this.properties = properties;
    }

    public void logQueryCount(QueryCountPerRequest queryCountPerRequest) {
        if (properties.getError().isEnable() && queryCountPerRequest.getTotalQueryCount() >= properties.getError().getCount()) {
            logger.error(LOG_MESSAGE_FORMAT, queryCountPerRequest.getApiUrl(), queryCountPerRequest.getTotalQueryCount(), queryCountPerRequest.getTotalQueryMilliSeconds());
        } else if (properties.getWarn().isEnable() && queryCountPerRequest.getTotalQueryCount() >= properties.getWarn().getCount()) {
            logger.warn(LOG_MESSAGE_FORMAT, queryCountPerRequest.getApiUrl(), queryCountPerRequest.getTotalQueryCount(), queryCountPerRequest.getTotalQueryMilliSeconds());
        } else if (properties.getInfo().isEnable() && queryCountPerRequest.getTotalQueryCount() >= properties.getInfo().getCount()) {
            logger.info(LOG_MESSAGE_FORMAT, queryCountPerRequest.getApiUrl(), queryCountPerRequest.getTotalQueryCount(), queryCountPerRequest.getTotalQueryMilliSeconds());
        }
    }
}

File: src/main/java/hyeon9mak/multidatasourcequerycounter/QueryCountLogger.kt
package hyeon9mak.multidatasourcequerycounter

import mu.KotlinLogging
import org.springframework.stereotype.Component

@Component
class QueryCountLogger(
    private val properties: QueryCounterLoggingLevelProperties,
) {
    private val logger = KotlinLogging.logger {}

    fun logQueryCount(queryCountPerRequest: QueryCountPerRequest) {
        if (properties.error.enable && queryCountPerRequest.totalQueryCount >= properties.error.count) {
            logger.error(
                LOG_MESSAGE_FORMAT,
                queryCountPerRequest.apiUrl,
                queryCountPerRequest.totalQueryCount,
                queryCountPerRequest.totalQueryMilliSeconds,
            )
        } else if (properties.warn.enable && queryCountPerRequest.totalQueryCount >= properties.warn.count) {
            logger.warn(
                LOG_MESSAGE_FORMAT,
                queryCountPerRequest.apiUrl,
                queryCountPerRequest.totalQueryCount,
                queryCountPerRequest.totalQueryMilliSeconds,
            )
        } else if (properties.info.enable && queryCountPerRequest.totalQueryCount >= properties.info.count) {
            logger.info(
                LOG_MESSAGE_FORMAT,
                queryCountPerRequest.apiUrl,
                queryCountPerRequest.totalQueryCount,
                queryCountPerRequest.totalQueryMilliSeconds,
            )
        }
    }

    companion object {
        private const val LOG_MESSAGE_FORMAT: String = "'{}' - totalQueryCount: {}, totalSpendTime: {}ms"
    }
}
